libirods_rule_engine_plugin-indexing.cpp

```diff
@@ -697,6 +740,151 @@ namespace {
         return retvalue.size() ? retvalue : "application/octet-stream";
     }
 
+    class PermsCalc {
+        std::map<std::string,std::string> users,groups;
+        std::map<std::string,std::list<std::string>> members;
+        std::map<std::string,int> user_entry;
+        std::multimap<int,std::string> user_perms, group_perms;
+
+        static std::map<int,std::string> perm_names;
+
+        bool is_group( const std::string &gid ) {
+            try {
+                groups.at(gid);
+            }
+            catch (const std::out_of_range&) {
+                return false;
+            }
+            return true;
+        }
+
+        bool is_member_of (const std::string& user_id, const std::string& group_id) {
+            try {
+                const auto & user_list = members.at(group_id);
+                return std::find( user_list.begin(), user_list.end(), user_id) != user_list.end();
+            }
+            catch (...) {
+              std::cerr << ("unknown error\n");
+              throw;
+            }
```

Comment:
is this what we want?
- `catch (...)`?
- `std::cerr`?
- rethrow?

Reply:
Hmm...

Reply:
replaced with irods::log style logging. (Caught when group_id doesn't exist as a group)